robotframework_reportportal/result_visitor.py

```diff
@@ -90,11 +91,12 @@ def end_test(self, test):
             'tags': list(test.tags),
             # for backward compatibility with Robot < 4.0 mark every test case
             # as critical if not set
-            'critical': getattr(test, 'critical', 'yes'),
+            'critical': getattr(test, 'critical', True),
```

Comment:
The same as for comment above. You are changing attribute type breaking the original logic behind it.